Surface3d example fails to compile

Output with 1.1rc3:

```
Compilation failed:

surface3d/surface3d.ts:41:19 - error TS2503: Cannot find namespace 'vis'.

41   private _graph: vis.Graph3d
                     ~~~
surface3d/surface3d.ts:53:23 - error TS2304: Cannot find name 'vis'.

53     this._graph = new vis.Graph3d(this.el, this.get_data(), this.model.options)
                         ~~~
surface3d/surface3d.ts:66:15 - error TS2503: Cannot find namespace 'vis'.

66   get_data(): vis.DataSet {
                 ~~~
surface3d/surface3d.ts:67:22 - error TS2304: Cannot find name 'vis'.

67     const data = new vis.DataSet()
                        ~~~
surface3d/surface3d.ts:119:17 - error TS1005: ';' expected.

119     this.define {
                    ~
2019-04-07 19:49:49,456 Uncaught exception
Traceback (most recent call last):
  File "/Users/bryan/anaconda/envs/11dev/lib/python3.7/site-packages/bokeh/util/compiler.py", line 338, in bundle_models
    _bundle_cache[key] = bundle = _bundle_models(custom_models)
  File "/Users/bryan/anaconda/envs/11dev/lib/python3.7/site-packages/bokeh/util/compiler.py", line 555, in _bundle_models
    custom_impls = _compile_models(custom_models)
  File "/Users/bryan/anaconda/envs/11dev/lib/python3.7/site-packages/bokeh/util/compiler.py", line 538, in _compile_models
    raise CompilationError(compiled.error)
bokeh.util.compiler.CompilationError:
surface3d/surface3d.ts:41:19 - error TS2503: Cannot find namespace 'vis'.

41   private _graph: vis.Graph3d
                     ~~~
surface3d/surface3d.ts:53:23 - error TS2304: Cannot find name 'vis'.

53     this._graph = new vis.Graph3d(this.el, this.get_data(), this.model.options)
                         ~~~
surface3d/surface3d.ts:66:15 - error TS2503: Cannot find namespace 'vis'.

66   get_data(): vis.DataSet {
                 ~~~
surface3d/surface3d.ts:67:22 - error TS2304: Cannot find name 'vis'.

67     const data = new vis.DataSet()
                        ~~~
surface3d/surface3d.ts:119:17 - error TS1005: ';' expected.

119     this.define {
                    ~
```
cc @mattpap

--- a/bokeh/models/widgets/icons.py
+++ b/bokeh/models/widgets/icons.py
@@ -27,7 +27,7 @@ log = logging.getLogger(__name__)
 # Bokeh imports
 from ...core.has_props import abstract
 
-from .widget import Widget
+from ...model import Model
 
 #-----------------------------------------------------------------------------
 # Globals and constants
@@ -46,7 +46,7 @@ __all__ = (
 #-----------------------------------------------------------------------------
 
 @abstract
-class AbstractIcon(Widget):
+class AbstractIcon(Model):
     ''' An abstract base class for icon widgets.
 
     '''

--- a/bokehjs/src/lib/models/widgets/abstract_button.ts
+++ b/bokehjs/src/lib/models/widgets/abstract_button.ts
@@ -47,7 +47,9 @@ export abstract class AbstractButtonView extends ControlView {
     const icon = this.model.icon
     if (icon != null) {
       build_views(this.icon_views, [icon], {parent: this})
-      prepend(this.button_el, this.icon_views[icon.id].el, nbsp())
+      const icon_view = this.icon_views[icon.id]
+      icon_view.render()
+      prepend(this.button_el, icon_view.el, nbsp())
     }
 
     this.group_el = div({class: "bk-btn-group"}, this.button_el)

--- a/bokehjs/src/lib/models/widgets/abstract_icon.ts
+++ b/bokehjs/src/lib/models/widgets/abstract_icon.ts
@@ -1,19 +1,20 @@
-import {Widget, WidgetView} from "./widget"
+import {Model} from "model"
+import {DOMView} from "core/dom_view"
 import * as p from "core/properties"
 
-export abstract class AbstractIconView extends WidgetView {
+export abstract class AbstractIconView extends DOMView {
   model: AbstractIcon
 }
 
 export namespace AbstractIcon {
   export type Attrs = p.AttrsOf<Props>
 
-  export type Props = Widget.Props
+  export type Props = Model.Props
 }
 
 export interface AbstractIcon extends AbstractIcon.Attrs {}
 
-export abstract class AbstractIcon extends Widget {
+export abstract class AbstractIcon extends Model {
   properties: AbstractIcon.Props
 
   constructor(attrs?: Partial<AbstractIcon.Attrs>) {

--- a/examples/app/surface3d/surface3d.ts
+++ b/examples/app/surface3d/surface3d.ts
@@ -13,6 +13,17 @@ import {HTMLBox, HTMLBoxView} from "models/layouts/html_box"
 import {ColumnDataSource} from "models/sources/column_data_source"
 import * as p from "core/properties"
 
+declare namespace vis {
+  class Graph3d {
+    constructor(el: HTMLElement, data: object, OPTIONS: object)
+    setData(data: vis.DataSet): void
+  }
+
+  class DataSet {
+    add(data: unknown): void
+  }
+}
+
 // This defines some default options for the Graph3d feature of vis.js
 // See: http://visjs.org/graph3d_examples.html for more details. This
 // JS object should match the Python default value.
@@ -116,13 +127,13 @@ export class Surface3d extends HTMLBox {
     // types have counterparts, e.g. ``bokeh.core.properties.String`` will be
     // ``p.String`` in the JS implementatin. Where the JS type system is not yet
     // as rich, you can use ``p.Any`` as a "wildcard" property type.
-    this.define {
-      x:           [ p.String           ]
-      y:           [ p.String           ]
-      z:           [ p.String           ]
-      data_source: [ p.Instance         ]
-      options:     [ p.Any,     OPTIONS ]
-    }
+    this.define<Surface3d.Props>({
+      x:           [ p.String           ],
+      y:           [ p.String           ],
+      z:           [ p.String           ],
+      data_source: [ p.Instance         ],
+      options:     [ p.Any,     OPTIONS ],
+    })
   }
 }
 Surface3d.initClass()

--- a/examples/custom/font-awesome/fontawesome.less
+++ b/examples/custom/font-awesome/fontawesome.less
@@ -1,27 +1,29 @@
-@font-awesome: "./node_modules/font-awesome";
+.bk-root {
+  @font-awesome: "./node_modules/font-awesome";
 
-@import "@{font-awesome}/less/variables.less";
-@fa-css-prefix: bk-u-fa;
-@import "@{font-awesome}/less/mixins.less";
+  @import "@{font-awesome}/less/variables.less";
+  @fa-css-prefix: bk-u-fa;
+  @import "@{font-awesome}/less/mixins.less";
 
-@font-face {
-  font-family: 'FontAwesome';
-  src: data-uri('@{font-awesome}/fonts/fontawesome-webfont.eot')   format('embedded-opentype'),
-       data-uri('@{font-awesome}/fonts/fontawesome-webfont.woff2') format('woff2'),
-       data-uri('@{font-awesome}/fonts/fontawesome-webfont.woff')  format('woff'),
-       data-uri('@{font-awesome}/fonts/fontawesome-webfont.ttf')   format('truetype'),
-       data-uri('@{font-awesome}/fonts/fontawesome-webfont.svg')   format('svg');
-  font-weight: normal;
-  font-style: normal;
+  @font-face {
+    font-family: 'FontAwesome';
+    src: data-uri('@{font-awesome}/fonts/fontawesome-webfont.eot')   format('embedded-opentype'),
+         data-uri('@{font-awesome}/fonts/fontawesome-webfont.woff2') format('woff2'),
+         data-uri('@{font-awesome}/fonts/fontawesome-webfont.woff')  format('woff'),
+         data-uri('@{font-awesome}/fonts/fontawesome-webfont.ttf')   format('truetype'),
+         data-uri('@{font-awesome}/fonts/fontawesome-webfont.svg')   format('svg');
+    font-weight: normal;
+    font-style: normal;
+  }
+
+  @import "@{font-awesome}/less/core.less";
+  @import "@{font-awesome}/less/larger.less";
+  @import "@{font-awesome}/less/fixed-width.less";
+  @import "@{font-awesome}/less/list.less";
+  @import "@{font-awesome}/less/bordered-pulled.less";
+  @import "@{font-awesome}/less/animated.less";
+  @import "@{font-awesome}/less/rotated-flipped.less";
+  @import "@{font-awesome}/less/stacked.less";
+  @import "@{font-awesome}/less/icons.less";
+  @import "@{font-awesome}/less/screen-reader.less";
 }
-
-@import "@{font-awesome}/less/core.less";
-@import "@{font-awesome}/less/larger.less";
-@import "@{font-awesome}/less/fixed-width.less";
-@import "@{font-awesome}/less/list.less";
-@import "@{font-awesome}/less/bordered-pulled.less";
-@import "@{font-awesome}/less/animated.less";
-@import "@{font-awesome}/less/rotated-flipped.less";
-@import "@{font-awesome}/less/stacked.less";
-@import "@{font-awesome}/less/icons.less";
-@import "@{font-awesome}/less/screen-reader.less";

--- a/examples/custom/font-awesome/fontawesome_icon.ts
+++ b/examples/custom/font-awesome/fontawesome_icon.ts
@@ -1,11 +1,9 @@
+import {AbstractIcon, AbstractIconView} from "models/widgets/abstract_icon"
 import * as p from "core/properties"
-
-import {AbstractIcon} from "models/widgets/abstract_icon"
-import {WidgetView} from "models/widgets/widget"
 
 import "./fontawesome.less"
 
-export class FontAwesomeIconView extends WidgetView {
+export class FontAwesomeIconView extends AbstractIconView {
   model: FontAwesomeIcon
 
   connect_signals(): void {
@@ -17,6 +15,7 @@ export class FontAwesomeIconView extends WidgetView {
     super.render()
 
     this.el.style.display = "inline"
+    this.el.style.verticalAlign = "middle"
     this.el.style.fontSize = `${this.model.size}em`
 
     this.el.classList.add("bk-u-fa")

--- a/tests/baselines/examples/custom/font-awesome/font-awesome
+++ b/tests/baselines/examples/custom/font-awesome/font-awesome
@@ -1,2 +1,2 @@
-Column bbox=[0, 0, 310, 41]
-  Button bbox=[5, 5, 300, 31]
+Column bbox=[0, 0, 310, 58]
+  Button bbox=[5, 5, 300, 48]

